src/beautifiers/markdown-blocks/index.coffee

```diff
@@ -0,0 +1,52 @@
+module.exports = (text, logger) ->
+  return new @Promise((resolve, reject) ->
+    extractCodeBlocks = require 'gfm-code-blocks'
+    beautifier = require '../../beautifier'
+    cleanMarkdown = text
+    codeBlocks = extractCodeBlocks(cleanMarkdown)
+    beautifyBlockPromises = []
+    for codeBlock in codeBlocks
+      codeLanguage = codeBlock.lang.match(/^[\w\d]+/)?[0]
+      logger.verbose "Beautify code block #{codeLanguage}"
+      if !codeLanguage
+        beautifyBlockPromises.push(null)
+      else
+        beautifyBlockPromises.push(new @Promise((resolve, reject) ->
+          linesInBlock = codeBlock.block.split('\n')
+          codePrefix = linesInBlock[0]
+          codeSuffix = linesInBlock[linesInBlock.length - 1]
+          code = linesInBlock.slice(1, linesInBlock.length - 1).join('\n')
+          filePath = 'fakepath.' + codeLanguage
+          grammar = atom.grammars.selectGrammar(filePath, code)
+          grammarName = grammar.name
+          allOptions = beautifier.getOptionsForPath(filePath)
```

Comment:
This is a fairly large code block. Please break it down into logically grouped functions.

For example:

```javascript
function languageForCodeBlock(codeBlock) {
  return codeBlock.lang.match(/^[\w\d]+/)?[0];
}
```

And 

```javascript
function allOptionsForLanguage(codeLanguage) {
  filePath = 'fakepath.' + codeLanguage
  grammar = atom.grammars.selectGrammar(filePath, code)
  grammarName = grammar.name
  return beautifier.getOptionsForPath(filePath)
}
```

etc.

Anything you can think of to make the code more readable and maintainable 😄 .

Reply:
I addressed the first example by creating a new file that focuses on language detection (file extension detection, to be precise). Extracting `allOptionsForLanguage` ended up being more challenging and I cancelled it. The problem is that the beautifier requires not only `allOptions`, but also the grammar and the path (`beautifier.beautify(code, allOptions, grammarName, filePath)`). Besides, the instance of `beautifier` is being used to extract options, meaning that this object would have to become a function argument. Too much wires to justify code splitting IMO.